data/src/apiKey/java/org/cryptomator/data/cloud/pcloud/PCloudImpl.kt

```diff
@@ -286,13 +285,73 @@ internal class PCloudImpl(context: Context, cloud: PCloud) {
 		}
 	}
 
+	private fun readFile(filePath : String, sink: DataSink, listener : ProgressListener) {
+		var attempts = 0
+		while (++attempts <= MaxContentLinkDownloadAttempts) {
+			val fileLink = apiClient.createFileLink(filePath, DownloadOptions.DEFAULT).execute()
+			try {
+				// Attempt to download the link's content, starting with the best link variant.
+				for (url in fileLink.urls()) {
+					try {
+						fileLink.download(url, sink, listener)
```

Comment:
```suggestion
						fileLink.download(url, sink, listener)
						break
```

We need to add a `break` directly after `fileLink.download(url, sink, listener)`, otherwise `fileLink.download(url, sink, listener)` will be executed for all `fileLink.urls()`, regardless of whether there was an error or not. If it was successful it should be executed only once.